app/lib/features/onboarding/providers/analytics_preferences_provider.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:acter/common/providers/common_providers.dart';
+import 'package:acter/common/providers/notifiers/client_pref_notifier.dart';
+import 'package:acter/config/setup.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:matomo_tracker/matomo_tracker.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+
+// Analytics preference keys
+const String canReportSentryKey = 'analytics.sentry';
+const String matomoAnalyticsKey = 'analytics.matomo';
+const String basicTelemetryKey = 'analytics.telemetry';
+const String researchKey = 'analytics.research';
+
+// Analytics preference providers
+final canReportSentryProvider = createAsyncPrefProvider<bool>(
+  prefKey: canReportSentryKey,
+  defaultValue: false,
+);
+
+final matomoAnalyticsProvider = createAsyncPrefProvider<bool>(
+  prefKey: matomoAnalyticsKey,
+  defaultValue: false,
+);
+
+final basicTelemetryProvider = createAsyncPrefProvider<bool>(
+  prefKey: basicTelemetryKey,
+  defaultValue: false,
+);
+
+final researchProvider = createAsyncPrefProvider<bool>(
+  prefKey: researchKey,
+  defaultValue: false,
+);
+
+Future<SentryEvent?> sentryBeforeSend(SentryEvent evt, Hint hint) async {
+  final canReport = await mainProviderContainer.read(canReportSentryProvider.future);
+  if (!canReport) {
+    return null;
+  }
+  return evt;
+}
+
+/// Handles Matomo-specific analytics settings
+Future<void> setMatomoAnalytics(bool value, WidgetRef ref) async {
+  // Update Matomo tracking state
+  MatomoTracker.instance.setOptOut(optOut: value);
```

Comment:
but isn't this the other way round? If `true` we are opt-in, but in this case it would do an opt-out on true.
```suggestion
  MatomoTracker.instance.setOptOut(optOut: !value);
```

Reply:
Added in this [commit](https://github.com/acterglobal/a3/pull/2801/commits/ebe1c7c5b208ef8f7256aff56a32b34912a3595b)